clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerSaslClient.java

```diff
@@ -88,7 +88,7 @@ public boolean hasInitialResponse() {
                         throw new SaslException("Expected empty challenge");
                     callbackHandler().handle(new Callback[] {callback});
                     setState(State.RECEIVE_SERVER_FIRST_MESSAGE);
-                    return String.format("n,,auth=Bearer %s", callback.token().value())
+                    return String.format("n,,\u0001auth=Bearer %s\u0001\u0001", callback.token().value())
```

Comment:
Could we try to reuse
https://github.com/apache/kafka/blob/03e788b29be6da0ef729f38448dc4345e68e398f/clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerSaslClient.java#L52
in this part of the code? I'm not sure how simple it is since I don't have extensive experience in Java, but reusing code and having explicit separators is always a good thing. If we could do it in the server as well, that'd be great